### PR TITLE
BUG: Fix issue with inserting duplicate columns in a dataframe (#14291)

### LINKED ISSUE #14291: BUG: DataFrame.insert with allow_duplicates=True fails when already duplicates present
upon DataFrame.insert option allow_duplicates works, but only only once.
When i have 2 columns with same name, additon of third throws

> ValueError: Wrong number of items passed 2, placement implies 1

#### Code Sample, a copy-pastable example if possible

```
a = pd.DataFrame()
a.insert(0, "qwe", [1,2,3,4], allow_duplicates=True)
a.insert(0, "qwe", [1,2,3,4], allow_duplicates=True)
a.insert(0, "qwe", [1,2,3,4], allow_duplicates=True)
```

#### Expected Output
   zxc  qwe  qwe
0    1    1    1
1    2    2    2
2    3    3    3
3    4    4    4

#### output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.12.final.0
python-bits: 64
OS: Windows
OS-release: 7
machine: AMD64
processor: Intel64 Family 6 Model 58 Stepping 9, GenuineIntel
byteorder: little
LC_ALL: None
LANG: ru_RU

pandas: 0.18.1
nose: 1.3.7
pip: 8.1.2
setuptools: 25.1.6
Cython: 0.24.1
numpy: 1.11.1
scipy: 0.18.0
statsmodels: 0.6.1
xarray: None
IPython: 5.1.0
sphinx: 1.4.6
patsy: 0.4.1
dateutil: 2.5.3
pytz: 2016.6.1
blosc: None
bottleneck: 1.0.0
tables: 3.2.2
numexpr: 2.6.1
matplotlib: 1.5.3
openpyxl: 2.3.2
xlrd: 1.0.0
xlwt: 1.1.2
xlsxwriter: 0.9.2
lxml: 3.6.4
bs4: 4.4.1
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 1.0.13
pymysql: 0.7.6.None
psycopg2: None
jinja2: 2.8
boto: 2.40.0
pandas_datareader: None
<\details>

### PR DESCRIPTION
**Pull request by paul-mannino**
- [x] closes #14291
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

Resubmitting PR (@jorisvandenbossche)

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -2590,9 +2590,24 @@ def assign(self, **kwargs):
...
+        """
+        Ensures new columns (which go into the BlockManager as new blocks) are
+        always copied and converted into an array.
+
+        Parameters
+        ----------
+        keys : object
+        value : int, Series, or array-like
+        broadcast : bool
+            Indicates whether all columns with the given key should be
```
Comment (jreback) [inline pandas/core/frame.py]: this is not obvious what you mean here. pls reword. (also this is not typically what ``broadcast`` means)

File: pandas/sparse/frame.py [hunk 2]
```diff
@@ -302,7 +302,28 @@ def fillna(self, value=None, method=None, axis=0, inplace=False,
...
 
-    def _sanitize_column(self, key, value):
+    def _sanitize_column(self, key, value, broadcast=True):
+        """
+        Creates a new SparseArray from the input value.
+
+        Parameters
+        ----------
+        keys : object
+        value : int, Series, or array-like
```
Comment (jreback) [inline pandas/sparse/frame.py]: same as above

File: pandas/sparse/frame.py [hunk 3]
```diff
@@ -302,7 +302,28 @@ def fillna(self, value=None, method=None, axis=0, inplace=False,
...
+            Indicates whether all columns with the given key should be
+            returned.
+
+        Returns
+        -------
+        sanitized_column : SparseArray
+
+        Notes
+        -----
+        The "broadcast" parameter was added to match the method signature of
```
Comment (paul-mannino) [inline pandas/sparse/frame.py]: Just to be clear, you want **kwargs for SparseDataFrame._sanitize_column and an optional parameter for DataFrame._sanitize_column? Should I add *args too so there's no risk of error?

``key`` isn't used in SparseDataFrame._sanitize_column either. Do you want me to get rid of that, too?

File: pandas/core/frame.py [hunk 4]
```diff
@@ -2590,9 +2590,25 @@ def assign(self, **kwargs):
...
+    def _sanitize_column(self, key, value, propagate=True):
+        """
+        Ensures new columns (which go into the BlockManager as new blocks) are
+        always copied and converted into an array.
+
+        Parameters
+        ----------
+        key : object
+        value : scalar, Series, or array-like
+        propagate : bool, default True
```
Comment (jreback) [inline pandas/core/frame.py]: no ``broadcast`` is the correct name. I just wanted the explanation to be what you have now.

**Comment by jorisvandenbossche** [comment]
This looks good to me (restarted the failing mac build on travis, should be unrelated)

**Comment by jorisvandenbossche** [comment]
@paul-mannino Can you rebase (there is a merge conflict, probably in the whatsnew file)

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**